File: CE303/src/MarketClientJava/ClientProgram.java
package MarketClientJava;

import java.util.Scanner;

public class ClientProgram {

    public static void main(String[] args) {
        int clientID;
            try {
                Scanner in = new Scanner(System.in); //user input

                try (Client client = new Client()) {
                    clientID = client.getClientID();
                    System.out.printf("Status: Connected with Client ID:%2d", clientID);
                    client.isActive();

                    while (true) {

                        int traderID = client.getTraderID(clientID);
                        int[] traders = client.getTraders(); //activeTraders List
                        int fromTrader;
                        int toTrader;

                        for (int t : traders) {
                            if (client.getStock(t)) {
                                System.out.printf("\n[Online] Trader %2d <- StockOwner.", t);
                            } else {
                                System.out.printf("\n[Online] Trader %2d", t);
                            }
                        }

                        if (client.getStock(traderID)) {
                            System.out.printf("\nTrader %2d: You are the Stock Owner.\n", traderID);
                            fromTrader = traderID;

                            System.out.println("\nPress <Enter Button> to update online traders. \nOr press 't' to Transfer stock to available traders.");
                            String next = in.nextLine().trim().toLowerCase();

                            if (next.equals("t")) {
                                System.out.println("\nEnter the Trader ID of who you want to transfer the stock to. \nYou can also transfer to yourself");
                                toTrader = Integer.parseInt(in.nextLine());
                                client.transferStock(fromTrader, toTrader, client.getStock(traderID));
                            }

                            else {
                                if (next.equals("")) {}//do nothing, run loop again
                                else
                                    throw new Exception("Unknown command " + next);
                            }
                        }
                        else {
                            System.out.printf("\nTrader %2d: You are not the Stock Owner.\n", traderID);
                            System.out.println("\nWaiting for Stock transfer...");
                            System.out.println("\nPress <Enter Button> to update online traders, and to check the Stock Status");
                            String next = in.nextLine().trim();
                            if (client.getStock(traderID)){next.equals(""); continue;} //if transferred stock, run loop again
                            if (next.equals("")) {} //do nothing - runs loop again, printing trader list
                            else {
                                throw new Exception("Unknown command " + next);
                            }

                        }
                    }
                }
            } catch (Exception e) {
                System.out.println(e.getMessage());
            }
        }
    }

File: CE303/MarketCS/MarketClientCS/ClientProgram.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Threading;

namespace MarketClientCS
{
    class ClientProgram
    {
        static void Main(string[] args)
        {
            int clientID;
            try
            {
                using (Client client = new Client())
                {
                    clientID = client.GetClientID();
                    Console.WriteLine("Status: Connected with Client ID: " + clientID);
                    client.IsActive();

                    while (true)
                    {
                        int traderID = client.GetTraderID(clientID);
                        int[] traders = client.GetTraders(); //activeTraders List
                        int fromTrader;
                        int toTrader;

                        foreach (int t in traders)
                        {
                            if (client.GetStock(t)){
                                Console.WriteLine($"[Online] Trader {t} <- StockOwner");
                            }
                            else
                            {
                                Console.WriteLine($"[Online] Trader {t}");
                            }
                        }
                        if (client.GetStock(traderID)) {
                            Console.WriteLine($"Trader {traderID}: You are the Stock Owner.");
                            fromTrader = traderID;

                            Console.WriteLine("Press <Enter Button> to update online traders. \nOr press 't' to Transfer stock to available traders.");
                            string next = Console.ReadLine().Trim().ToLower();

                            if (next.Equals("t")) {
                                Console.WriteLine("Enter the Trader ID of who you want to transfer the stock to. \nYou can also transfer to yourself");
                                toTrader = int.Parse(Console.ReadLine());
                                client.TransferStock(fromTrader, toTrader, client.GetStock(traderID));
                            }

                            else {
                                if (next.Equals("")) {}//do nothing, run loop again
                                else
                                    throw new Exception("Unknown command " + next);
                            }
                        }
                        else {
                            Console.WriteLine($"Trader {traderID}: You are not the Stock Owner.");
                            Console.WriteLine("Waiting for Stock transfer...");
                            Console.WriteLine("Press <Enter Button> to update online traders, and to check the Stock Status");
                            string next = Console.ReadLine().Trim();
                            if (client.GetStock(traderID)){next.Equals(""); continue;} //if transferred stock, run loop again
                            if (next.Equals("")) {} //do nothing - runs loop again, printing trader list
                            else {
                                throw new Exception("Unknown command " + next);
                            }
                        }
                    }
                }
            } catch (Exception e){
                Console.WriteLine(e.Message);
            }
        }
    }
}